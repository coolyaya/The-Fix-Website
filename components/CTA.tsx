import Link from "next/link";

export function CTA() {
  return (
    <section className="bg-fix-blue text-white">
      <div className="container flex flex-col items-start gap-6 py-16 md:flex-row md:items-center md:justify-between md:py-24">
        <div className="space-y-3">
          <h2 className="text-3xl font-semibold md:text-4xl">Ready to Get Your Fix?</h2>
          <p className="text-lg text-white/80">Walk in or book online in under a minute.</p>
        </div>
        <Link
          href="/support"
          className="inline-flex items-center rounded-xl bg-fix-pink px-6 py-3 text-base font-semibold text-white shadow-soft transition hover:opacity-90 focus:outline-none focus:ring-4 focus:ring-fix-blue/30"
        >
          Book a Repair
        </Link>
      </div>
    </section>
  );
}
