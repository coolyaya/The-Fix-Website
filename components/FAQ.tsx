const FAQ_ITEMS = [
  {
    question: "How long will my repair take?",
    answer:
      "Most screen and battery repairs are completed the same day\u2014often under an hour.",
  },
  {
    question: "Do you use quality parts?",
    answer:
      "Yes. We use high-quality parts and back our work with a parts & labor warranty.",
  },
  {
    question: "Do I need an appointment?",
    answer:
      "Walk-ins welcome. Booking online guarantees the fastest service.",
  },
] as const;

export function FAQ() {
  return (
    <section>
      <div className="container space-y-8 py-16 md:space-y-10 md:py-24">
        <div className="max-w-2xl space-y-4">
          <h2 className="text-3xl font-semibold md:text-4xl">Frequently asked questions</h2>
          <p className="text-lg text-fix-slate">
            Answers to the top questions we hear every day. Need something else? Reach out and our team will get back fast.
          </p>
        </div>
        <div className="grid gap-4">
          {FAQ_ITEMS.map(({ question, answer }) => (
            <div key={question} className="rounded-2xl bg-white p-6 shadow-soft">
              <h3 className="text-xl font-semibold text-fix-navy">{question}</h3>
              <p className="mt-2 text-sm text-fix-slate">{answer}</p>
            </div>
          ))}
        </div>
      </div>
    </section>
  );
}
