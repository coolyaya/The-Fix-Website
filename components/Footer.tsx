"use client";

import { FormEvent, useState } from "react";
import Link from "next/link";
import { Facebook, Instagram, Mail, Phone, Youtube } from "lucide-react";
import { Button } from "@/components/ui/button";
import { Input } from "@/components/ui/input";

export function Footer() {
  const [submitted, setSubmitted] = useState(false);

  function handleSubmit(event: FormEvent<HTMLFormElement>) {
    event.preventDefault();
    setSubmitted(true);
    const form = event.currentTarget;
    form.reset();
  }

  return (
    <footer className="border-t border-border/60 bg-muted/30">
      <div className="container grid gap-10 py-12 md:grid-cols-3 md:gap-16">
        <div className="space-y-4">
          <h3 className="text-lg font-semibold">The Fix</h3>
          <p className="max-w-md text-sm text-muted-foreground">
            Your neighborhood tech repair crew since 2011. Certified technicians, premium parts,
            and same-day turnarounds for phones, tablets, laptops, and more.
          </p>
          <div className="space-y-1 text-sm text-muted-foreground">
            <p className="flex items-center gap-2">
              <Phone className="h-4 w-4" aria-hidden="true" />
              <a href="tel:15551234567" className="hover:text-fix-blue">
                (555) 123-4567
              </a>
            </p>
            <p className="flex items-center gap-2">
              <Mail className="h-4 w-4" aria-hidden="true" />
              <a href="mailto:hello@thefix.com" className="hover:text-fix-blue">
                hello@thefix.com
              </a>
            </p>
            <p>Hours: Mon-Sat 10-7, Sun 11-5</p>
          </div>
          <div className="flex items-center gap-3 text-muted-foreground">
            <Link href="#" aria-label="Follow The Fix on Instagram" className="hover:text-fix-blue">
              <Instagram className="h-5 w-5" />
            </Link>
            <Link href="#" aria-label="Follow The Fix on Facebook" className="hover:text-fix-blue">
              <Facebook className="h-5 w-5" />
            </Link>
            <Link href="#" aria-label="Subscribe to The Fix on YouTube" className="hover:text-fix-blue">
              <Youtube className="h-5 w-5" />
            </Link>
          </div>
        </div>

        <div className="grid grid-cols-2 gap-6 text-sm">
          <div className="space-y-3">
            <p className="text-sm font-semibold text-foreground">Navigate</p>
            <ul className="space-y-2 text-muted-foreground">
              <li>
                <Link href="/" className="hover:text-fix-blue">
                  Home
                </Link>
              </li>
              <li>
                <Link href="/services" className="hover:text-fix-blue">
                  Services
                </Link>
              </li>
              <li>
                <Link href="/locations" className="hover:text-fix-blue">
                  Locations
                </Link>
              </li>
              <li>
                <Link href="/about" className="hover:text-fix-blue">
                  About
                </Link>
              </li>
              <li>
                <Link href="/support" className="hover:text-fix-blue">
                  Support
                </Link>
              </li>
            </ul>
          </div>
          <div className="space-y-3">
            <p className="text-sm font-semibold text-foreground">Services</p>
            <ul className="space-y-2 text-muted-foreground">
              <li>Screen repair</li>
              <li>Battery replacement</li>
              <li>Water damage</li>
              <li>Accessory upgrades</li>
              <li>Device protection</li>
            </ul>
          </div>
        </div>

        <div className="space-y-4">
          <p className="text-sm font-semibold text-foreground">Stay in the loop</p>
          <p className="text-sm text-muted-foreground">
            Get promos, new accessories, and repair tips monthly. We never spam.
          </p>
          <form className="flex flex-col gap-3 sm:flex-row" onSubmit={handleSubmit}>
            <Input
              type="email"
              name="email"
              placeholder="Email address"
              aria-label="Email address"
              required
              className="w-full"
            />
            <Button type="submit" className="rounded-xl">
              Subscribe
            </Button>
          </form>
          {submitted ? (
            <p className="text-sm font-medium text-fix-blue">Thanks! We'll keep you posted.</p>
          ) : (
            <p className="text-xs text-muted-foreground">Mocked opt-in. No emails sent.</p>
          )}
        </div>
      </div>
      <div className="border-t border-border/40 bg-background/40 py-6 text-center text-xs text-muted-foreground">
        &copy; {new Date().getFullYear()} The Fix. All rights reserved.
      </div>
    </footer>
  );
}


