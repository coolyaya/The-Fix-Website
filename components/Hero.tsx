import Image from "next/image";
import Link from "next/link";
import { Check } from "lucide-react";

import { Button } from "@/components/ui/button";

export function Hero() {
  return (
    <section className="bg-gradient-to-b from-fix-sky to-fix-white">
      <div className="container grid gap-12 py-16 md:grid-cols-[minmax(0,1fr)_minmax(0,520px)] md:items-center md:py-24">
        <div className="space-y-6">
          <span className="inline-flex w-fit items-center gap-2 rounded-full bg-white px-4 py-1 text-xs font-semibold uppercase tracking-wide text-fix-blue shadow-soft">
            Device emergencies solved daily
          </span>
          <h1 className="text-4xl font-semibold leading-tight md:text-5xl">
            Fast, Expert Phone & Device Repair
          </h1>
          <p className="text-lg text-fix-slate md:text-xl">
            Cracked screen? Battery dying? Get same-day repairs backed by warranty{"\u2014"}done by certified technicians.
          </p>
          <div className="flex flex-wrap gap-3">
            <Button asChild size="lg">
              <Link href="/support">Book a Repair</Link>
            </Button>
            <Button asChild variant="outline" size="lg">
              <Link href="/support#estimate">Get an Instant Estimate</Link>
            </Button>
          </div>
          <ul className="grid gap-3 text-sm text-fix-slate md:grid-cols-2">
            {[
              "Same-day turnarounds on the repairs you need most",
              "Certified techs using premium, device-safe tools",
              "Backed by parts and labor warranty",
              "Walk in anytime or schedule online in under a minute",
            ].map((item) => (
              <li key={item} className="flex items-start gap-2">
                <Check className="mt-0.5 h-4 w-4 text-fix-pink" aria-hidden />
                <span>{item}</span>
              </li>
            ))}
          </ul>
        </div>

        <div className="relative aspect-[4/3] overflow-hidden rounded-3xl border border-fix-blue/10 bg-white shadow-soft">
          <Image
            src="/promos/iphone17-case.jpg"
            alt="Certified technician installing a new iPhone display"
            fill
            sizes="(max-width: 768px) 100vw, 520px"
            className="object-cover"
            priority
          />
          <div className="pointer-events-none absolute inset-x-6 bottom-6 rounded-2xl bg-white/90 p-4 shadow-soft">
            <p className="text-xs font-semibold uppercase tracking-wide text-fix-blue">
              Most repairs in under an hour
            </p>
            <p className="text-sm text-fix-slate">
              Book online and we'll have parts ready before you arrive.
            </p>
          </div>
        </div>
      </div>
    </section>
  );
}

