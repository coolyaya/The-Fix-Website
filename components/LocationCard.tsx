"use client";

import Image from "next/image";
import { useEffect, useState } from "react";
import { MapPinned, Phone } from "lucide-react";

import { Button } from "@/components/ui/button";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { cn, formatMiles } from "@/lib/utils";

export interface Location {
  id: string;
  name: string;
  address: string;
  lat: number;
  lng: number;
  phone: string;
  hours: string;
  photos: string[];
}

interface LocationCardProps {
  location: Location;
  distanceMiles?: number;
  isActive?: boolean;
  onHover?: (location: Location | null) => void;
  onSelect?: (location: Location) => void;
}

const AUTO_ROTATE_MS = 4000;

export function LocationCard({
  location,
  distanceMiles,
  isActive = false,
  onHover,
  onSelect,
}: LocationCardProps) {
  const [photoIndex, setPhotoIndex] = useState(0);

  useEffect(() => {
    if (!location.photos.length) return;
    const id = window.setInterval(() => {
      setPhotoIndex((current) => (current + 1) % location.photos.length);
    }, AUTO_ROTATE_MS);
    return () => window.clearInterval(id);
  }, [location.photos.length]);

  const directionsHref =
    "https://www.google.com/maps/dir/?api=1&destination=" +
    encodeURIComponent(location.address);

  return (
    <Card
      onMouseEnter={() => onHover?.(location)}
      onMouseLeave={() => onHover?.(null)}
      onFocus={() => onHover?.(location)}
      onBlur={() => onHover?.(null)}
      className={cn(
        "h-full cursor-pointer rounded-2xl border border-fix-blue/10 bg-white shadow-soft transition-all hover:-translate-y-1 hover:shadow-lg focus-visible:outline-none focus-visible:ring-4 focus-visible:ring-fix-blue/30",
        isActive && "border-fix-blue",
      )}
      onClick={() => onSelect?.(location)}
      tabIndex={0}
      role="button"
      aria-pressed={isActive}
    >
      <div className="relative h-40 w-full overflow-hidden rounded-t-2xl">
        {location.photos.length ? (
          <Image
            key={location.photos[photoIndex]}
            src={location.photos[photoIndex]}
            alt={`${location.name} storefront`}
            fill
            sizes="(max-width: 768px) 100vw, 33vw"
            className="object-cover"
          />
        ) : (
          <div className="flex h-full w-full items-center justify-center bg-fix-blue/5 text-sm text-fix-slate">
            Photo coming soon
          </div>
        )}
        <div className="absolute bottom-3 left-3 flex gap-1">
          {location.photos.map((photo, index) => (
            <span
              key={photo}
              className={cn(
                "h-2 w-2 rounded-full bg-white/50",
                index === photoIndex && "bg-white",
              )}
            />
          ))}
        </div>
      </div>
      <CardHeader>
        <CardTitle className="flex flex-col gap-1 text-lg text-fix-navy">
          <span>{location.name}</span>
          {distanceMiles !== undefined && (
            <span className="text-sm font-medium text-fix-blue">
              {formatMiles(distanceMiles)} away
            </span>
          )}
        </CardTitle>
        <p className="text-sm text-fix-slate">{location.address}</p>
      </CardHeader>
      <CardContent className="space-y-3 text-sm text-fix-slate">
        <p className="flex items-center gap-2 font-medium text-fix-navy">
          <Phone className="h-4 w-4 text-fix-blue" aria-hidden />
          <a href={`tel:${location.phone}`} className="hover:text-fix-blue">
            {location.phone}
          </a>
        </p>
        <p>{location.hours}</p>
        <Button asChild variant="outline" size="sm" className="w-full">
          <a href={directionsHref} target="_blank" rel="noopener noreferrer">
            <MapPinned className="mr-2 h-4 w-4" />
            Get directions
          </a>
        </Button>
      </CardContent>
    </Card>
  );
}




