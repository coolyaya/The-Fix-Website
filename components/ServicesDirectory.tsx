"use client";

import React, { useMemo, useState } from "react";

import { Badge } from "@/components/ui/badge";
import { Card, CardContent, CardHeader, CardTitle } from "@/components/ui/card";
import { Input } from "@/components/ui/input";
import {
  Select,
  SelectContent,
  SelectItem,
  SelectTrigger,
  SelectValue,
} from "@/components/ui/select";
import { cn } from "@/lib/utils";
import type { Service } from "@/lib/services";

const collator = new Intl.Collator("en", { sensitivity: "base" });
const currencyFormatter = new Intl.NumberFormat("en-US", {
  style: "currency",
  currency: "USD",
  minimumFractionDigits: 0,
  maximumFractionDigits: 2,
});

interface ServicesDirectoryProps {
  services: Service[];
}

export function ServicesDirectory({ services }: ServicesDirectoryProps) {
  const [searchValue, setSearchValue] = useState("");
  const [categoryFilter, setCategoryFilter] = useState("all");

  const categories = useMemo(() => {
    const unique = new Set(services.map((service) => service.category));
    return [...unique].sort((a, b) => collator.compare(a, b));
  }, [services]);

  const normalizedSearch = searchValue.trim().toLowerCase();

  const groupedServices = useMemo(() => {
    const matches = services.filter((service) => {
      if (categoryFilter !== "all" && service.category !== categoryFilter) {
        return false;
      }

      if (!normalizedSearch) return true;

      const options = service.variants ?? service.models ?? [];
      const haystack = [
        service.name,
        service.category,
        service.description,
        service.slug,
        ...options.map(getOptionLabel),
      ]
        .join(" ")
        .toLowerCase();

      return haystack.includes(normalizedSearch);
    });

    const grouped = new Map<string, Service[]>();
    matches.forEach((service) => {
      if (!grouped.has(service.category)) {
        grouped.set(service.category, []);
      }
      grouped.get(service.category)?.push(service);
    });

    const sortedEntries = [...grouped.entries()].map(([category, items]) => {
      const sortedItems = [...items].sort((a, b) => {
        if (a.featured !== b.featured) {
          return a.featured ? -1 : 1;
        }
        return collator.compare(a.name, b.name);
      });
      return [category, sortedItems] as const;
    });

    sortedEntries.sort((a, b) => collator.compare(a[0], b[0]));
    return sortedEntries;
  }, [services, categoryFilter, normalizedSearch]);

  return (
    <section className="space-y-8">
      <div className="flex flex-col gap-4 lg:flex-row lg:items-end lg:justify-between">
        <div className="flex flex-1 flex-col gap-2">
          <label className="sr-only" htmlFor="services-search">
            Search services
          </label>
          <Input
            id="services-search"
            value={searchValue}
            onChange={(event) => setSearchValue(event.target.value)}
            placeholder="Search by repair, device, or keyword"
            aria-label="Search services"
            className="w-full"
          />
        </div>
        <div className="flex flex-col gap-2 lg:w-60">
          <label className="sr-only" htmlFor="services-category">
            Filter by category
          </label>
          <Select value={categoryFilter} onValueChange={setCategoryFilter}>
            <SelectTrigger id="services-category" aria-label="Filter by category">
              <SelectValue placeholder="All categories" />
            </SelectTrigger>
            <SelectContent>
              <SelectItem value="all">All categories</SelectItem>
              {categories.map((category) => (
                <SelectItem key={category} value={category}>
                  {category}
                </SelectItem>
              ))}
            </SelectContent>
          </Select>
        </div>
      </div>

      {groupedServices.length === 0 ? (
        <div className="rounded-2xl border border-dashed border-border/70 bg-muted/30 p-10 text-center text-sm text-muted-foreground">
          No services match your filters. Try adjusting the search or category.
        </div>
      ) : (
        <div className="space-y-12">
          {groupedServices.map(([category, items]) => (
            <CategorySection key={category} category={category} services={items} />
          ))}
        </div>
      )}
    </section>
  );
}

interface CategorySectionProps {
  category: string;
  services: Service[];
}

function CategorySection({ category, services }: CategorySectionProps) {
  const sectionId = `category-${category.toLowerCase().replace(/[^a-z0-9]+/g, "-").replace(/(^-|-$)/g, "")}`;

  return (
    <section aria-labelledby={`${sectionId}-title`} className="space-y-6" id={sectionId}>
      <div className="sticky top-24 z-10 flex flex-col gap-1 border-b border-border/60 bg-background/95 py-3 backdrop-blur supports-[backdrop-filter]:bg-background/80">
        <h2 id={`${sectionId}-title`} className="text-2xl font-semibold text-foreground">
          {category}
        </h2>
        <p className="text-sm text-muted-foreground">
          {services.length} {services.length === 1 ? "service" : "services"}
        </p>
      </div>
      <div className="grid gap-6 md:grid-cols-2 xl:grid-cols-3">
        {services.map((service) => (
          <ServiceCard key={service.slug} service={service} />
        ))}
      </div>
    </section>
  );
}

interface ServiceCardProps {
  service: Service;
}

function ServiceCard({ service }: ServiceCardProps) {
  const options = service.variants ?? service.models ?? [];
  const hasMultipleOptions = options.length > 1;
  const [selectedIndex, setSelectedIndex] = useState(0);

  const selectedOption = options[selectedIndex] ?? options[0];
  const selectLabel = service.variants ? "Choose option" : "Choose model";

  return (
    <Card
      id={service.slug}
      role="article"
      aria-labelledby={`${service.slug}-title`}
      className={cn(
        "group relative flex h-full flex-col justify-between border-border/60",
        service.featured && "border-fix-blue/80 shadow-[0_0_0_1.5px_rgba(0,106,255,0.3)]"
      )}
    >
      <CardHeader className="space-y-4">
        <div className="flex items-start justify-between gap-3">
          <div className="space-y-2">
            <CardTitle id={`${service.slug}-title`} className="text-lg font-semibold text-foreground">
              {service.name}
            </CardTitle>
            <p className="text-sm text-muted-foreground">{service.description}</p>
          </div>
          {service.featured ? (
            <Badge variant="secondary" className="rounded-full bg-fix-blue/10 text-fix-blue">
              Featured
            </Badge>
          ) : null}
        </div>
        <div className="flex flex-wrap gap-2 text-xs">
          <Badge variant="outline" className="rounded-full border-border/50 text-muted-foreground">
            ~{service.duration_min} min
          </Badge>
          <Badge variant="outline" className="rounded-full border-border/50 text-muted-foreground">
            {service.warranty_days}-day warranty
          </Badge>
        </div>
      </CardHeader>
      <CardContent className="mt-auto space-y-4">
        <div className="rounded-2xl border border-border/50 bg-muted/40 p-4">
          <div className="text-xs uppercase tracking-wide text-muted-foreground">Current price</div>
          <div className="mt-1 text-2xl font-semibold text-foreground" aria-live="polite">
            {formatPrice(selectedOption.price)}
          </div>
          {hasMultipleOptions ? (
            <div className="mt-4">
              <Select value={String(selectedIndex)} onValueChange={(value) => setSelectedIndex(Number(value))}>
                <SelectTrigger aria-label={selectLabel} className="w-full">
                  <SelectValue placeholder={selectLabel} />
                </SelectTrigger>
                <SelectContent>
                  {options.map((option, index) => (
                    <SelectItem key={`${service.slug}-${index}`} value={String(index)}>
                      {getOptionLabel(option)} - {formatPrice(option.price)}
                    </SelectItem>
                  ))}
                </SelectContent>
              </Select>
            </div>
          ) : (
            <p className="mt-2 text-sm text-muted-foreground">{getOptionLabel(selectedOption)}</p>
          )}
        </div>
      </CardContent>
    </Card>
  );
}

function getOptionLabel(
  option: Service["variants"][number] | Service["models"][number]
): string {
  return "option" in option ? option.option : option.model;
}

function formatPrice(price: number) {
  return currencyFormatter.format(price);
}
