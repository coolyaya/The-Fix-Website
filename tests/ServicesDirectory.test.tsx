import React from "react";
import { render, screen, within } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { describe, expect, it } from "vitest";

import { ServicesDirectory } from "@/components/ServicesDirectory";
import type { Service } from "@/lib/services";

const sampleServices: Service[] = [
  {
    category: "Screens",
    name: "iPhone 14 Screen Replacement",
    variants: [
      { option: "OEM", price: 249 },
      { option: "Aftermarket", price: 179 },
    ],
    duration_min: 90,
    warranty_days: 180,
    description: "Genuine OLED replacement",
    featured: true,
    slug: "iphone-14-screen-replacement",
  },
  {
    category: "Batteries",
    name: "iPad 9th Gen Battery Service",
    models: [
      { model: "Wi-Fi", price: 119 },
      { model: "Cellular", price: 139 },
    ],
    duration_min: 60,
    warranty_days: 365,
    description: "High cycle count battery swap",
    featured: false,
    slug: "ipad-9-battery-service",
  },
];

describe("ServicesDirectory", () => {
  it("renders cards and updates price when a variant is selected", async () => {
    const user = userEvent.setup();
    const { container } = render(<ServicesDirectory services={sampleServices} />);

    expect(screen.getAllByRole("article")).toHaveLength(sampleServices.length);
    expect(container).toMatchSnapshot("initial-services-directory");

    const iPhoneCard = screen.getByRole("article", { name: "iPhone 14 Screen Replacement" });
    expect(within(iPhoneCard).getByText("$249")).toBeInTheDocument();

    const trigger = within(iPhoneCard).getByRole("combobox", { name: "Choose option" });
    await user.click(trigger);

    const aftermarketOption = await screen.findByRole("option", { name: "Aftermarket - $179" });
    await user.click(aftermarketOption);

    expect(within(iPhoneCard).getByText("$179")).toBeInTheDocument();
  });
});

