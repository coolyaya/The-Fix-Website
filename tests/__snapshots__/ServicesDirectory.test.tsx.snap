// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ServicesDirectory > renders cards and updates price when a variant is selected > initial-services-directory 1`] = `
<div>
  <section
    class="space-y-8"
  >
    <div
      class="flex flex-col gap-4 lg:flex-row lg:items-end lg:justify-between"
    >
      <div
        class="flex flex-1 flex-col gap-2"
      >
        <label
          class="sr-only"
          for="services-search"
        >
          Search services
        </label>
        <input
          aria-label="Search services"
          class="flex h-9 rounded-md border border-input bg-transparent px-3 py-1 text-base shadow-sm transition-colors file:border-0 file:bg-transparent file:text-sm file:font-medium file:text-foreground placeholder:text-muted-foreground focus-visible:outline-none focus-visible:ring-1 focus-visible:ring-ring disabled:cursor-not-allowed disabled:opacity-50 md:text-sm w-full"
          id="services-search"
          placeholder="Search by repair, device, or keyword"
          value=""
        />
      </div>
      <div
        class="flex flex-col gap-2 lg:w-60"
      >
        <label
          class="sr-only"
          for="services-category"
        >
          Filter by category
        </label>
        <button
          aria-autocomplete="none"
          aria-controls="radix-:r0:"
          aria-expanded="false"
          aria-label="Filter by category"
          class="flex h-9 w-full items-center justify-between whitespace-nowrap rounded-md border border-input bg-transparent px-3 py-2 text-sm shadow-sm ring-offset-background data-[placeholder]:text-muted-foreground focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1"
          data-state="closed"
          dir="ltr"
          id="services-category"
          role="combobox"
          type="button"
        >
          <span
            style="pointer-events: none;"
          >
            All categories
          </span>
          <svg
            aria-hidden="true"
            class="lucide lucide-chevron-down h-4 w-4 opacity-50"
            fill="none"
            height="24"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="24"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="m6 9 6 6 6-6"
            />
          </svg>
        </button>
      </div>
    </div>
    <div
      class="space-y-12"
    >
      <section
        aria-labelledby="category-batteries-title"
        class="space-y-6"
        id="category-batteries"
      >
        <div
          class="sticky top-24 z-10 flex flex-col gap-1 border-b border-border/60 bg-background/95 py-3 backdrop-blur supports-[backdrop-filter]:bg-background/80"
        >
          <h2
            class="text-2xl font-semibold text-foreground"
            id="category-batteries-title"
          >
            Batteries
          </h2>
          <p
            class="text-sm text-muted-foreground"
          >
            1
             
            service
          </p>
        </div>
        <div
          class="grid gap-6 md:grid-cols-2 xl:grid-cols-3"
        >
          <div
            aria-labelledby="ipad-9-battery-service-title"
            class="rounded-xl border bg-card text-card-foreground shadow group relative flex h-full flex-col justify-between border-border/60"
            id="ipad-9-battery-service"
            role="article"
          >
            <div
              class="flex flex-col p-6 space-y-4"
            >
              <div
                class="flex items-start justify-between gap-3"
              >
                <div
                  class="space-y-2"
                >
                  <div
                    class="tracking-tight text-lg font-semibold text-foreground"
                    id="ipad-9-battery-service-title"
                  >
                    iPad 9th Gen Battery Service
                  </div>
                  <p
                    class="text-sm text-muted-foreground"
                  >
                    High cycle count battery swap
                  </p>
                </div>
              </div>
              <div
                class="flex flex-wrap gap-2 text-xs"
              >
                <div
                  class="inline-flex items-center border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 rounded-full border-border/50 text-muted-foreground"
                >
                  ~
                  60
                   min
                </div>
                <div
                  class="inline-flex items-center border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 rounded-full border-border/50 text-muted-foreground"
                >
                  365
                  -day warranty
                </div>
              </div>
            </div>
            <div
              class="p-6 pt-0 mt-auto space-y-4"
            >
              <div
                class="rounded-2xl border border-border/50 bg-muted/40 p-4"
              >
                <div
                  class="text-xs uppercase tracking-wide text-muted-foreground"
                >
                  Current price
                </div>
                <div
                  aria-live="polite"
                  class="mt-1 text-2xl font-semibold text-foreground"
                >
                  $119
                </div>
                <div
                  class="mt-4"
                >
                  <button
                    aria-autocomplete="none"
                    aria-controls="radix-:r1:"
                    aria-expanded="false"
                    aria-label="Choose model"
                    class="flex h-9 items-center justify-between whitespace-nowrap rounded-md border border-input bg-transparent px-3 py-2 text-sm shadow-sm ring-offset-background data-[placeholder]:text-muted-foreground focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 w-full"
                    data-state="closed"
                    dir="ltr"
                    role="combobox"
                    type="button"
                  >
                    <span
                      style="pointer-events: none;"
                    >
                      Wi-Fi
                       - 
                      $119
                    </span>
                    <svg
                      aria-hidden="true"
                      class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m6 9 6 6 6-6"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
      <section
        aria-labelledby="category-screens-title"
        class="space-y-6"
        id="category-screens"
      >
        <div
          class="sticky top-24 z-10 flex flex-col gap-1 border-b border-border/60 bg-background/95 py-3 backdrop-blur supports-[backdrop-filter]:bg-background/80"
        >
          <h2
            class="text-2xl font-semibold text-foreground"
            id="category-screens-title"
          >
            Screens
          </h2>
          <p
            class="text-sm text-muted-foreground"
          >
            1
             
            service
          </p>
        </div>
        <div
          class="grid gap-6 md:grid-cols-2 xl:grid-cols-3"
        >
          <div
            aria-labelledby="iphone-14-screen-replacement-title"
            class="rounded-xl border bg-card text-card-foreground group relative flex h-full flex-col justify-between border-fix-blue/80 shadow-[0_0_0_1.5px_rgba(0,106,255,0.3)]"
            id="iphone-14-screen-replacement"
            role="article"
          >
            <div
              class="flex flex-col p-6 space-y-4"
            >
              <div
                class="flex items-start justify-between gap-3"
              >
                <div
                  class="space-y-2"
                >
                  <div
                    class="tracking-tight text-lg font-semibold text-foreground"
                    id="iphone-14-screen-replacement-title"
                  >
                    iPhone 14 Screen Replacement
                  </div>
                  <p
                    class="text-sm text-muted-foreground"
                  >
                    Genuine OLED replacement
                  </p>
                </div>
                <div
                  class="inline-flex items-center border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 border-transparent hover:bg-secondary/80 rounded-full bg-fix-blue/10 text-fix-blue"
                >
                  Featured
                </div>
              </div>
              <div
                class="flex flex-wrap gap-2 text-xs"
              >
                <div
                  class="inline-flex items-center border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 rounded-full border-border/50 text-muted-foreground"
                >
                  ~
                  90
                   min
                </div>
                <div
                  class="inline-flex items-center border px-2.5 py-0.5 text-xs font-semibold transition-colors focus:outline-none focus:ring-2 focus:ring-ring focus:ring-offset-2 rounded-full border-border/50 text-muted-foreground"
                >
                  180
                  -day warranty
                </div>
              </div>
            </div>
            <div
              class="p-6 pt-0 mt-auto space-y-4"
            >
              <div
                class="rounded-2xl border border-border/50 bg-muted/40 p-4"
              >
                <div
                  class="text-xs uppercase tracking-wide text-muted-foreground"
                >
                  Current price
                </div>
                <div
                  aria-live="polite"
                  class="mt-1 text-2xl font-semibold text-foreground"
                >
                  $249
                </div>
                <div
                  class="mt-4"
                >
                  <button
                    aria-autocomplete="none"
                    aria-controls="radix-:r2:"
                    aria-expanded="false"
                    aria-label="Choose option"
                    class="flex h-9 items-center justify-between whitespace-nowrap rounded-md border border-input bg-transparent px-3 py-2 text-sm shadow-sm ring-offset-background data-[placeholder]:text-muted-foreground focus:outline-none focus:ring-1 focus:ring-ring disabled:cursor-not-allowed disabled:opacity-50 [&>span]:line-clamp-1 w-full"
                    data-state="closed"
                    dir="ltr"
                    role="combobox"
                    type="button"
                  >
                    <span
                      style="pointer-events: none;"
                    >
                      OEM
                       - 
                      $249
                    </span>
                    <svg
                      aria-hidden="true"
                      class="lucide lucide-chevron-down h-4 w-4 opacity-50"
                      fill="none"
                      height="24"
                      stroke="currentColor"
                      stroke-linecap="round"
                      stroke-linejoin="round"
                      stroke-width="2"
                      viewBox="0 0 24 24"
                      width="24"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="m6 9 6 6 6-6"
                      />
                    </svg>
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </section>
    </div>
  </section>
</div>
`;
